src/main.cpp

```diff
@@ -24,7 +27,23 @@ int main(int /*argc*/, char const* const* /*argv*/)
     print_thread_result(valik_thread_result);
     // carts should be (bin, [reads])
 
-    cart_queue queue{nr_bins, cart_capacity};
+    cart_queue queue{nr_bins, cart_capacity, max_carts_queued};
+
+    auto threads = std::vector<std::jthread>{};
+
+    // spanws a few worker threads that consume the data
+    for (size_t i{0}; i < 4; ++i) {
+        threads.emplace_back([&]() {
+            auto next = queue.dequeue();
+            while (next) {
+                // !TODO what should be done with the full cart?
+                // write *next to file
+                // call system("stellar.....");
+                next = queue.dequeue();
+            }
+        });
+    }
+
 
     for (auto & query_result : valik_thread_result)
```

Comment:
This should be before the consumer part right?

Reply:
The consumer will get stuck in `dequeue`, but that is not a problem. The other way around the producers would get stuck in `insert`, but will stay there for ever, since no consumers will empty the full carts.